tasks/ssh.yml

```diff
@@ -31,6 +31,15 @@
       line: "X11Forwarding {{ security_ssh_x11_forwarding }}"
   notify: restart ssh
 
+- name: Ensure cloud-init conf does not still allow passwords anyway.
+  lineinfile:
+    dest: "{{ security_ssh_cloudinit_config_path }}"
+    regexp: "^PasswordAuthentication"
+    state: absent
```

Comment:
The way this is set up, it would still remove this option even if Password Auth is allowed, and it might also break on non-Ubuntu systems...

Reply:
```
- name: Apply same SSH configuration to Ubuntu cloud-init SSH configuration
  lineinfile:
    dest: "{{ security_ssh_cloud_init_config_path }}"
    regexp: "^PasswordAuthentication"
    line: "PasswordAuthentication {{ security_ssh_password_authentication }}"
    state: present
    validate: 'sshd -T -f %s'
    mode: 0600
  notify: restart ssh
  when: ansible_distribution == 'Ubuntu'
```

Would that be a better approach?

Reply:
I changed the Debian.yml to

```
---
security_ssh_config_path: /etc/ssh/sshd_config
security_ssh_config_dir_path: /etc/ssh/sshd_config.d
security_ssh_cloud_init_config_path: "{{ security_ssh_config_dir_path }}/50-cloud-init.conf"
security_sshd_name: ssh
```